### PR TITLE
feat: gitlab pipeline convert

### PR DESCRIPTION
**Pull request by mappjzc**
# Summary
Add gitlab ConvertPipelines.

### Does this close any open issues?
#2583 

### Screenshots
image
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/gitlab/tasks/pipeline_convertor.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			domainPipeline := &devops.CICDPipeline{
+				DomainEntity: domainlayer.DomainEntity{
+					Id: pipelineIdGen.Generate(data.Options.ConnectionId, gitlabPipeline.GitlabId),
+				},
+
+				Name:      fmt.Sprintf("%d", gitlabPipeline.GitlabId),
+				CommitSha: gitlabPipeline.Sha,
+				Branch:    gitlabPipeline.Ref,
+				Repo:      fmt.Sprintf("%d", gitlabPipeline.ProjectId),
+				Result:    gitlabPipeline.Status,
```
Comment (warren830) [inline plugins/gitlab/tasks/pipeline_convertor.go:87]: after pr #2683 get merged, please modify Result and Status, these two values should be enum

File: plugins/gitlab/tasks/pipeline_convertor.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+			Table: RAW_USER_TABLE,
+		},
+		Convert: func(inputRow interface{}) ([]interface{}, error) {
+			gitlabPipeline := inputRow.(*gitlabModels.GitlabPipeline)
+
+			createdAt := time.Now()
+			if gitlabPipeline.GitlabCreatedAt != nil {
+				createdAt = *gitlabPipeline.GitlabCreatedAt
+			}
+			finishedAt := time.Now()
```
Comment (warren830) [inline plugins/gitlab/tasks/pipeline_convertor.go:72]: what if finishedAt should be nil?

File: plugins/gitlab/tasks/pipeline_convertor.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+			Params: GitlabApiParams{
+				ConnectionId: data.Options.ConnectionId,
+				ProjectId:    data.Options.ProjectId,
+			},
+			Table: RAW_USER_TABLE,
+		},
+		Convert: func(inputRow interface{}) ([]interface{}, error) {
+			gitlabPipeline := inputRow.(*gitlabModels.GitlabPipeline)
+
+			createdAt := time.Now()
```
Comment (warren830) [inline plugins/gitlab/tasks/pipeline_convertor.go:68]: why not use createdDate?

**Review by warren830 — CHANGES_REQUESTED** [review]

**Comment by mappjzc** [comment]
closed because it code move to #2682

**Pull request closed without merging**